tests/test_convert_threaded.py

```diff
@@ -102,41 +102,16 @@ def test_convert_s3_path_sqlite_join(
         # sequential s3 SQLite files. See below for more information
         # https://cloudpathlib.drivendata.org/stable/caching/#automatically
         local_cache_dir=f"{fx_tempdir}/sqlite_s3_cache/2",
-        # note: we use a custom join to limit the
-        # data processing required within the context
-        # of GitHub Actions runner image resources.
-        joins="""
-            SELECT
-                image.Image_TableNumber,
-                image.Metadata_ImageNumber,
-                image.Metadata_Plate,
-                image.Metadata_Well,
-                image.Image_Metadata_Site,
-                image.Image_Metadata_Row,
-                cytoplasm.* EXCLUDE (Metadata_ImageNumber),
-                cells.* EXCLUDE (Metadata_ImageNumber),
-                nuclei.* EXCLUDE (Metadata_ImageNumber)
-            FROM
-                (SELECT * FROM read_parquet('cytoplasm.parquet') LIMIT 5000) AS cytoplasm
-            LEFT JOIN (SELECT * FROM read_parquet('cells.parquet') LIMIT 5000) AS cells ON
-                cells.Metadata_ImageNumber = cytoplasm.Metadata_ImageNumber
-                AND cells.Metadata_ObjectNumber = cytoplasm.Cytoplasm_Parent_Cells
-            LEFT JOIN (SELECT * FROM read_parquet('nuclei.parquet') LIMIT 5000) AS nuclei ON
-                nuclei.Metadata_ImageNumber = cytoplasm.Metadata_ImageNumber
-                AND nuclei.Metadata_ObjectNumber = cytoplasm.Cytoplasm_Parent_Nuclei
-            LEFT JOIN (SELECT * FROM read_parquet('image.parquet') LIMIT 5000) AS image ON
-                image.Metadata_ImageNumber = cytoplasm.Metadata_ImageNumber
-        """,
```

Comment:
Was this deleted because of the pagination fix?

Reply:
Thanks - yes, this is removed to defer to the preset default SQL for joins. The change here is mostly to remove `LIMIT 5000` from each joined table (from Parquet datasets) and instead use the full dataset. With this change we are now able to process the full dataset within the scope of a [GitHub Actions runner image's resources](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories) (4 CPU, 16 GB RAM). Earlier changes towards offset-based pagination resulted in #214 and an inability to run the full data through this test. Keyset pagination implemented in this PR is much more performant and doesn't require us to artificially limit the query here anymore.